### PR TITLE
fix!: unify resolvePath, always resolve request via resolver and remove resolve from fileProcessor

### PR DESCRIPTION
**Pull request by barak007**
Make all stylable import resolution for `.st.css` and `.js` and other extensions go through the resolver in a consistent way. 

- [x] fix resolve extensions options
- [x] fix cli build for linked projects
- [x] fix cli test-kit messages 

Breaking: 
* Stylable resolution does not accept extensions from integrations (webpack) it uses `require.extensions`
* Resolver `resolvePath` signature changed
* `TransformerOptions` added required `moduleResolver`
* Removed `timedCacheOptions` from Stylable
* Stylable `resolvePath` signature changed
* Stylable `process` signature changed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "watch": "yarn build -w",
     "lint": "eslint . -f codeframe",
     "pretest": "yarn build",
-    "test": "mocha \"./packages/*/dist/test/**/*.spec.js\" --parallel --timeout 10000",
+    "test": "mocha \"./packages/*/dist/test/**/*.spec.js\" --parallel --retry 3 --timeout 10000",
```
Comment (AviVahl) [inline package.json:13]: ??

Comment (barak007) [reply package.json:13]: just checking

File: packages/cli/test/test-kit/cli-test-kit.ts [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import { readdirSync, readFileSync, statSync, writeFileSync, existsSync, mkdirSync } from 'fs';
-import { spawn, ChildProcessWithoutNullStreams, spawnSync } from 'child_process';
```
Comment (idoros) [inline packages/cli/test/test-kit/cli-test-kit.ts:2]: changes in `cli-test-kit` are not directly related to the this PR. maybe add some comment in the PR description to indicate this change

File: packages/cli/src/build.ts [hunk 3]
```diff
@@ -95,12 +95,16 @@ export async function build({
     diagnostics,
     diagnosticsMode,
 }: BuildOptions) {
-    const { join, resolve } = fs;
```
Comment (idoros) [inline packages/cli/src/build.ts:98]: changes here also seems unrelated to PR

File: packages/core/src/module-resolver.ts [hunk 4]
```diff
@@ -8,8 +8,10 @@ import type { MinimalFS } from './cached-process-file';
 const resolverContext = {};
 
 export function createDefaultResolver(fileSystem: MinimalFS, resolveOptions: any): ModuleResolver {
+    const extensions = [...new Set([...(resolveOptions?.extensions ?? []), '.js'])];
```
[the remark this answers is not in the document]
Comment (barak007) [reply packages/core/src/module-resolver.ts:11]: it will become first...

**Comment by tzachbon** [comment]
This looks nice, left some comments.

**Review by tzachbon — COMMENTED** [review]
Looks nice, left some comments

File: packages/core/src/cached-process-file.ts [hunk 5]
```diff
@@ -8,47 +8,43 @@ export interface CacheItem<T> {
...
             !cached ||
             (cached && cached.stat.mtime.valueOf() !== stat.mtime.valueOf())
         ) {
-            const content = fs.readFileSync(resolvedPath, 'utf8');
-            const value = processContent(content, resolvedPath);
-
-            cache[resolvedPath] = { value, stat: { mtime: stat.mtime } };
+            const content = fs.readFileSync(fullpath, 'utf8');
+            const value = processContent(content, fullpath);
+            cache[fullpath] = { value, stat: { mtime: stat.mtime } };
```
[the remark this answers is not in the document]
Comment (tzachbon) [reply packages/core/src/cached-process-file.ts:38]: @barak007  this is important for the CLI

File: packages/core/src/index.ts [hunk 6]
```diff
@@ -96,7 +96,7 @@ export {
     valueMapping,
     valueParserWarnings,
 } from './stylable-value-parsers';
-export { StylableInfrastructure, createInfrastructure } from './create-infra-structure';
+export { createStylableFileProcessor } from './create-stylable-processor';
```
[the remark this answers is not in the document]
Comment (barak007) [reply packages/core/src/index.ts:99]: not really.
